### PR TITLE
DOC Clarify thumbnail_size and note change in default

### PR DESCRIPTION
**Pull request by lucyleeow**
towards #717 

Warn about change in `thumbnail_size` default.

### CODE REVIEW

File: sphinx_gallery/gen_gallery.py [hunk 1]
```diff
@@ -132,6 +132,12 @@ def _complete_gallery_conf(sphinx_gallery_conf, src_dir, plot_gallery,
             backreferences_warning,
             type=DeprecationWarning)
 
+    if gallery_conf.get("thumbnail_size", False):
```
Comment (larsoner) [inline sphinx_gallery/gen_gallery.py:135]:
```suggestion
    if 'thumbnail_size' not in gallery_conf:
```

Comment (larsoner) [reply sphinx_gallery/gen_gallery.py:135]: Oh wait, this actually needs to be `''thumbnail_size' not in sphinx_gallery_conf`. `gallery_conf` will always get a value above from `gallery_conf = copy.deepcopy(DEFAULT_GALLERY_CONF)`

File: doc/conf.py [hunk 2]
```diff
@@ -368,6 +368,7 @@ def setup(app):
     # each code block
     'capture_repr': ('_repr_html_', '__repr__'),
     'matplotlib_animations': True,
+    # To prevent depracation warning
```
Comment (larsoner) [inline doc/conf.py:371]: cruft

**Comment by lucyleeow** [comment]
@larsoner are you happy with new default being 2xCSS (320,224) and to mention it in the next release notes?

**Comment by larsoner** [comment]
Yeah I think it makes sense

File: CHANGES.rst [hunk 3]
```diff
@@ -1,6 +1,13 @@
 Change Log
 ==========
 
+v0.8.0
+------
+
+The default for configuration `thumbnail_size` will change from `(400, 280)`
+to `(320, 224)` in version 0.9.0.
+
```
Comment (larsoner) [inline CHANGES.rst:9]: Probably worth explaining that this is 2x the CSS size

**Comment by lucyleeow** [comment]
The errors seem to be due to failure to install mayavi:
```
 Building wheel for mayavi (setup.py) ... error
 ERROR: Command errored out with exit status 1:

   command: /home/travis/miniconda/envs/test/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-wvnui7ma/mayavi/setup.py'"'"'; __file__='"'"'/tmp/pip-install-wvnui7ma/mayavi/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-r_omtas9

       cwd: /tmp/pip-install-wvnui7ma/mayavi/

  Complete output (33 lines):

  running bdist_wheel

  running build

  Traceback (most recent call last):

    File "<string>", line 1, in <module>

    File "/tmp/pip-install-wvnui7ma/mayavi/setup.py", line 474, in <module>

      **config

    File "/home/travis/miniconda/envs/test/lib/python3.6/site-packages/numpy/distutils/core.py", line 171, in setup

      return old_setup(**new_attr)

    File "/home/travis/miniconda/envs/test/lib/python3.6/site-packages/setuptools/__init__.py", line 161, in setup

      return distutils.core.setup(**attrs)

    File "/home/travis/miniconda/envs/test/lib/python3.6/distutils/core.py", line 148, in setup

      dist.run_commands()

    File "/home/travis/miniconda/envs/test/lib/python3.6/distutils/dist.py", line 955, in run_commands

      self.run_command(cmd)

    File "/home/travis/miniconda/envs/test/lib/python3.6/distutils/dist.py", line 974, in run_command

      cmd_obj.run()

    File "/home/travis/miniconda/envs/test/lib/python3.6/site-packages/wheel/bdist_wheel.py", line 223, in run

      self.run_command('build')

    File "/home/travis/miniconda/envs/test/lib/python3.6/distutils/cmd.py", line 313, in run_command

      self.distribution.run_command(command)

    File "/home/travis/miniconda/envs/test/lib/python3.6/distutils/dist.py", line 974, in run_command

      cmd_obj.run()

    File "/tmp/pip-install-wvnui7ma/mayavi/setup.py", line 268, in run

      build_tvtk_classes_zip()

    File "/tmp/pip-install-wvnui7ma/mayavi/setup.py", line 254, in build_tvtk_classes_zip

      gen_tvtk_classes_zip()

    File "tvtk/setup.py", line 83, in gen_tvtk_classes_zip

      from tvtk.code_gen import TVTKGenerator

    File "/tmp/pip-install-wvnui7ma/mayavi/tvtk/code_gen.py", line 10, in <module>

      import vtk_module as vtk

    File "tvtk/vtk_module.py", line 15, in <module>

      from vtk import *

  ModuleNotFoundError: No module named 'vtk'

  ----------------------------------------

  ERROR: Failed building wheel for mayavi
```
Maybe @GaelVaroquaux or @larsoner you know more ... ? `vtk` is included in our install packages...?

**Comment by larsoner** [comment]
Thanks @lucyleeow

**Pull request merged**